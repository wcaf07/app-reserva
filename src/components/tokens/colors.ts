const colors = {
  blueLight: '#A4C64A',
  greenLight: '#46A2F8',
  white: '#ffffff',
};

export default colors;
