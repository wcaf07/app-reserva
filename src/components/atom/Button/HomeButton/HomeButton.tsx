import * as React from 'react';
import styled from 'styled-components/native';
import colors from '@tokens/colors';

type HomeButtonType = {
  bgColor: string;
  text: string;
};

const Container = styled.TouchableOpacity`
  height: 100px;
  background-color: ${(props: {bgColor: any}) => props.bgColor};
  align-items: center;
  justify-content: center;
`;

const Text = styled.Text`
  color: ${colors.white};
  font-weight: bold;
  font-size: 16px;
`;

const HomeButton: React.FC<HomeButtonType> = (atts) => {
  return (
    <Container bgColor={atts.bgColor}>
      <Text>{atts.text}</Text>
    </Container>
  );
};

export default HomeButton;
