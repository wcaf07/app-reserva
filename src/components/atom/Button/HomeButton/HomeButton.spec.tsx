import React from 'react';
import {render} from '@testing-library/react-native';
import HomeButton from './HomeButton';
import colors from '@/components/tokens/colors';

describe('HomeButton.styles', () => {
  test('THEN HomeButton SHOULD match with snapshot when is show', () => {
    const component = render(
      <HomeButton text="test" bgColor={colors.blueLight} />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
