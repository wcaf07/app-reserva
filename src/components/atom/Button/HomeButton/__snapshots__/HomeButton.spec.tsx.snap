// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeButton.styles THEN HomeButton SHOULD match with snapshot when is show 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#46A2F8",
      "height": 100,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
